airflow/providers/amazon/aws/operators/batch.py

```diff
@@ -19,114 +19,77 @@
 #
 
 """
-Airflow operator for AWS batch service
+An Airflow operator for AWS batch services
 
-.. seealso:: http://boto3.readthedocs.io/en/latest/reference/services/batch.html
-"""
-
-import sys
-from random import randint
-from time import sleep
-from typing import Optional
+.. seealso::
 
-import botocore.exceptions
-import botocore.waiter
+    - http://boto3.readthedocs.io/en/latest/guide/configuration.html
+    - http://boto3.readthedocs.io/en/latest/reference/services/batch.html
+    - https://docs.aws.amazon.com/batch/latest/APIReference/Welcome.html
+"""
 
-from airflow.contrib.hooks.aws_hook import AwsHook
 from airflow.exceptions import AirflowException
 from airflow.models import BaseOperator
-from airflow.typing_compat import Protocol
+from airflow.providers.amazon.aws.hooks.batch_client import AwsBatchClient
 from airflow.utils.decorators import apply_defaults
 
-# pylint: disable=invalid-name, unused-argument
 
-
-class BatchProtocol(Protocol):
```

Comment:
Removing this class seems to be not a backward compatible change. If that's true please add note in UPDATING.md. Additionally, maybe it is possible to preserve backward compatibility?

Reply:
This PR should have backward-compatible changes, although some code has moved from the `operator.batch` module into the `hooks.batch_client` module.  AFAIK, the `BatchProtocol` is only a typing extension that is not actually used in "consumer land", but I doesn't hurt to add some notes about it to the `UPDATING.md`.  The refactoring in this PR is essentially `extract-class` and `move-class` refactoring and it applied to previously private-methods that should all continue to behave the same way they did before (only the OOP design and packaging are improved and explicit now).  Sorry if that wasn't clear from the description.  (Sometimes scrolling through a github web page is not the best way to do a review, because it only offers piecemeal changes while scrolling down the web page; it's a good interface but still leaves a lot to be desired.)